docs/src/description/constraints.md

```diff
@@ -2,8 +2,158 @@
 
 ## Boundary constraints (`boundary_constraints`)
 
-TODO
+The `boundary_constraints` section consists of expressions describing the expected value of columns in the main or auxiliary traces at the specified boundary. Column boundaries can be selected using boundary accessors. Valid boundary accessors are `.first`, which selects the first cell of the column to which it is applied, and `.last`, which selects the last cell of the column column to which it is applied.
+
+**Boundary constraints are optional.** It is equally valid to define an empty `boundary_constraints` section or to omit the `boundary_constraints` section entirely.
```

Comment:
I think this is fine for now, but in the future we should change this. A computation without any boundary constraints is kind of meaningless - so, we should make them mandatory.

Reply:
I added #44